### PR TITLE
[3d-tiles]: Add CesiumIonLoader to allow resolving authorization in preload

### PR DESCRIPTION
**Pull request by xintongxia**
For #676

### CODE REVIEW

File: examples/deck.gl/3d-tiles/app.js [hunk 1]
```diff
@@ -108,14 +110,7 @@ export default class App extends PureComponent {
...
   async _loadExample(selectedExample) {
-    const {ionAssetId, ionAccessToken} = selectedExample;
-    if (ionAssetId && ionAccessToken) {
-      const metadata = await _getIonTilesetMetadata(ionAccessToken, ionAssetId);
-      const {url, headers} = metadata;
-      selectedExample.tilesetUrl = url;
-      selectedExample.headers = headers;
-      selectedExample.metadata = metadata;
-    }
+    selectedExample.tilesetUrl = `${TILESET_SERVER_URL}/${selectedExample.ionAssetId}/tileset.json`;
```
Comment (Pessimistress) [inline examples/deck.gl/3d-tiles/app.js]: - This is no longer async
- Don't mutate constants - just construct this URL in `render`

File: examples/deck.gl/3d-tiles/tile-3d-layer.js [hunk 2]
```diff
@@ -69,13 +69,17 @@ export default class Tile3DLayer extends CompositeLayer {
 
   async _loadTileset(tilesetUrl) {
```
Comment (Pessimistress) [inline examples/deck.gl/3d-tiles/tile-3d-layer.js]: We should remove this class from examples and use deck.gl's Tile3DLayer

File: modules/3d-tiles/src/index.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 // LOADERS
 export {default as Tiles3DLoader} from './tiles-3d-loader';
+export {default as IONLoader} from './ion-loader';
```
Comment (Pessimistress) [inline modules/3d-tiles/src/index.js]: `CesiumIonLoader`?

File: modules/3d-tiles/src/ion-loader.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import Tiles3DLoader from './tiles-3d-loader';
+import {getIonTilesetMetadata} from './lib/ion/ion';
+
+async function preload(url, options) {
```
Comment (Pessimistress) [inline modules/3d-tiles/src/ion-loader.js]: `options` is a nested object for all loader modules: `image`, `fetch`, etc.

File: examples/deck.gl/3d-tiles/tile-3d-layer.js (hunk 2 above)
Comment (xintongxia) [reply examples/deck.gl/3d-tiles/tile-3d-layer.js]: Yes, I always keep them sync and will delete it before this PR is landed.

File: modules/3d-tiles/src/ion-loader.js (hunk 4 above)
Comment (xintongxia) [reply modules/3d-tiles/src/ion-loader.js]: The nested options is tricky, that means I need copy options['cesium-ion'] to/from options['3d-tiles'].

**Pull request merged**